src/middlewared/middlewared/plugins/disk_/identify_freebsd.py

```diff
@@ -68,32 +64,32 @@ def identifier_to_device(self, ident, disks=None):
                 return search.text
 
         elif tp == 'serial':
-            search = geom.class_by_name('DISK').xml.find(
-                f'.//provider/config[ident = "{value}"]/../../name'
-            )
+            xml = geom.class_by_name('DISK').xml
+            search = xml.find(f'.//provider/config[ident = "{value}"]/../../name')
             if search is not None:
                 return search.text
-            # Builtin xml xpath do not understand normalize-space
-            search = etree.fromstring(ElementTree.tostring(geom.class_by_name('DISK').xml))
-            search = search.xpath(
-                './/provider/config['
-                f'normalize-space(ident) = normalize-space("{value}")'
-                ']/../../name'
-            )
-            if len(search) > 0:
-                return search[0].text
+
+            # remove leading/trailing and more than single whitespace char(s)
+            _value = ' '.join(value.split())
+            search = xml.find(f'.//provider/config[ident = "{_value}"]/../../name')
```

Comment:
Are we sure that XML document values are always whitespace-normalized?

Reply:
It's probably not, so I've fixed the logic.